src/core/Form/TextInput/TextInput.tsx

```diff
@@ -64,8 +71,31 @@ const StyledTextInput = styled(
  */
 export class TextInput extends Component<TextInputProps> {
   render() {
-    const { ...passProps } = withSuomifiDefaultProps(this.props);
+    const {
+      children,
+      icon,
+      iconProps,
+      className,
+      ...passProps
+    } = withSuomifiDefaultProps(this.props);
+
+    const prioritizedIcon = icon || iconProps?.icon;
+
+    const newIconProps = {
+      ...iconProps,
+      icon: prioritizedIcon,
+    };
```

Comment:
This part is not the cleanest, but without destructuring icon and iconProps out of the props an empty iconProps object was passed to the component.

Reply:
I think this is completely fine, just don't like the name of the icon that gets resolved from props and IconProps.

Reply:
Changed the name to resolvedIcon.